src/metabase/pulse.clj

```diff
@@ -373,22 +377,96 @@
 (defmethod send-notification! :slack
   [{:keys [channel-id message attachments]}]
   (let [attachments (create-and-upload-slack-attachments! attachments)]
-    (slack/post-chat-message! channel-id message attachments)))
+    (try
+      (slack/post-chat-message! channel-id message attachments)
+      (catch ExceptionInfo e
+        ;; Token errors have already been logged and we should not retry.
+        (when-not (contains? (:errors (ex-data e)) :slack-token)
+          (throw e))))))
 
 (defmethod send-notification! :email
   [{:keys [subject recipients message-type message]}]
-  (email/send-message!
-    :subject      subject
-    :recipients   recipients
-    :message-type message-type
-    :message      message))
+  (try
+    (email/send-message-or-throw! {:subject      subject
+                                   :recipients   recipients
+                                   :message-type message-type
+                                   :message      message})
+    (catch ExceptionInfo e
+      (when (not= :smtp-host-not-set (:cause (ex-data e)))
+        (throw e)))))
+
+(declare ^:private reconfigure-retrying)
+
+(defsetting notification-retry-max-attempts
+  (deferred-tru "The maximum number of attempts for delivering a single notification.")
+  :type :integer
+  :default 7
+  :on-change reconfigure-retrying)
+
+(defsetting notification-retry-initial-interval
+  (deferred-tru "The initial retry delay in milliseconds when delivering notifications.")
+  :type :integer
+  :default 500
+  :on-change reconfigure-retrying)
+
+(defsetting notification-retry-multiplier
+  (deferred-tru "The delay multiplier between attempts to deliver a single notification.")
+  :type :double
+  :default 2.0
+  :on-change reconfigure-retrying)
+
+(defsetting notification-retry-randomizaion-factor
```

Comment:
@metamben 
We missed this misspelling. If we fix it, can we drop the old setting, so it won't cause problems during Serialization (unknown settings causes errors on load)?

randomizaion => randomization

Reply:
oh good catch. We can fix the spelling of the setting and then
- if the correctly spelled version is null, look for the misspelled version
- if present and its not an env variable value, clear it and set the correctly spelled version
- if its an env variable we cannot clear its value. We can warn that we've fixed the misspelling and they should fix the mispelling in their startup command for metabase

Reply:
If we fix it soon-ish, then I think it's okay to completely ignore the old spelling in envvar, since it's currently only active on 44.0, so I don't think a lot has used this envvar yet. That should simplify the fix too.